freestyle-config/src/main/scala/config.scala

```diff
@@ -74,6 +78,10 @@ object config {
         def empty: M[Config] = ME.pure(loadConfig(ConfigFactory.empty()))
         def parseString(s: String): M[Config] =
           ME.catchNonFatal(loadConfig(ConfigFactory.parseString(s)))
+        def loadAs[T](decoder: ConfigDecoder[T]): M[Either[DecodeError, T]] =
```

Comment:
We don't need to return `Either[DecodeError, T]]` in these methods since we are in the context of monad error.
We can just return `T` and fold on case classy's result to send the Left to `raiseError` and the Right to `pure`.